### ISSUE TITLE
Don't save every tile ID in data/data2D on a new line

### ISSUE THREAD

**Issue by codecat** [Ogmo-Editor-3/OgmoEditor3-CE#52]
When opening a level file, I see this:

![image](https://user-images.githubusercontent.com/136534/69007400-9246f400-093d-11ea-95b1-c9bb9929c3fe.png)

It's a lot of wasted bytes, especially since it's all spaces! I'd suggest keeping the readability of the json file while "compressing" these tile indices together, so that it's still readable. Perhaps something like this:

![image](https://user-images.githubusercontent.com/136534/69007450-2a44dd80-093e-11ea-97c5-d6b6ea80a43c.png)

And additionally split it into multiple chunks like the Tiled editor does: (but perhaps that can be different issue altogether)

![image](https://user-images.githubusercontent.com/136534/69007463-55c7c800-093e-11ea-88db-2b93e1f3fc29.png)

**Comment by AustinEast** [comment]
ah yeah, we’ve been meaning to get to this 😅 i’ve attempted to format the exported tiles based on the map’s width/height in the past, but it ended up being fairly non-trivial, so i dropped it in favor of other stuff. 

ill take another wack at it when i get the chance 👍

**Comment by codecat** [comment]
Perhaps a good feature as well would be to allow saving the json file without the pretty print entirely, but maybe that's already implemented?

**Comment by AustinEast** [comment]
alrighty, I cleaned up the pretty printed JSON as much as I could (in commit 35356518c1cd36bf8c9236cd1c101b312a99c9fe). its not perfect, but it handles the huge arrays a lot better I think. If we want prettier printing, someone else may have to take a stab at it!

in addition, I added an option to the project editing panel to pick whether you want pretty or compact printing, for when you want the smallest filesize possible and don't care about how it looks :)
